packages/components/src/test/table/UseRows.test.tsx

```diff
@@ -242,4 +248,113 @@ describe("useRows", () => {
+
+    const { result } = renderHook((props: UseRowsProps) => useRows(props), { initialProps: { ...initialProps, pageSize: 10 } });
+
+    await waitFor(() => {
+      expect(result.current.rows).to.have.lengthOf(0);
+      expect(result.current.isLoading).to.be.false;
+    });
+
+    onActiveFormattingUnitSystemChanged.raiseEvent({ system: "metric" });
+
+    await waitFor(() => {
+      expect(result.current.rows).to.have.lengthOf(0);
+      presentationManagerMock.verify(async (x) => x.getContentAndSize(moq.It.isAny()), moq.Times.once());
+    });
+  });
+
+  it("reloads rows in pages", async () => {
+    const propertiesField = createTestPropertiesContentField({
+      name: "first_field",
+      label: "First Field",
+      properties: [{ property: createTestPropertyInfo() }],
+    });
+    const descriptor = createTestContentDescriptor({ fields: [propertiesField] });
+    const itemsCount = ROWS_RELOAD_PAGE_SIZE + 1;
+    const items = Array.from(Array(itemsCount).keys()).map((i) =>
+      createTestContentItem({
+        values: { [propertiesField.name]: `test_value_${i}` },
+        displayValues: { [propertiesField.name]: `Test value ${i}` },
+      }),
+    );
+
+    presentationManagerMock
+      .setup(async (x) => x.getContentAndSize(moq.It.is(({ paging }) => paging?.start === 0 && paging?.size === itemsCount)))
+      .returns(async () => ({ content: new Content(descriptor, items), size: itemsCount }));
+
+    // all items should be loaded with single request
+    const { result } = renderHook((props: UseRowsProps) => useRows(props), { initialProps: { ...initialProps, pageSize: itemsCount } });
+    await waitFor(() => {
+      expect(result.current.rows).to.have.lengthOf(itemsCount);
+    });
+
+    // setup presentation manager for rows reload
+    presentationManagerMock
+      .setup(async (x) => x.getContentAndSize(moq.It.is(({ paging }) => paging?.start === 0 && paging?.size === ROWS_RELOAD_PAGE_SIZE)))
+      .returns(async () => ({ content: new Content(descriptor, items.slice(0, ROWS_RELOAD_PAGE_SIZE)), size: ROWS_RELOAD_PAGE_SIZE }));
+    presentationManagerMock
+      .setup(async (x) => x.getContentAndSize(moq.It.is(({ paging }) => paging?.start === ROWS_RELOAD_PAGE_SIZE && paging?.size === 1)))
+      .returns(async () => ({ content: new Content(descriptor, items.slice(ROWS_RELOAD_PAGE_SIZE)), size: 1 }));
+
+    onActiveFormattingUnitSystemChanged.raiseEvent({ system: "metric" });
+
+    await waitFor(() => {
+      expect(result.current.rows).to.have.lengthOf(itemsCount);
+      presentationManagerMock.verify(
+        async (x) => x.getContentAndSize(moq.It.is(({ paging }) => paging?.start === 0 && paging?.size === ROWS_RELOAD_PAGE_SIZE)),
+        moq.Times.once(),
+      );
+      presentationManagerMock.verify(
+        async (x) => x.getContentAndSize(moq.It.is(({ paging }) => paging?.start === ROWS_RELOAD_PAGE_SIZE && paging?.size === 1)),
+        moq.Times.once(),
+      );
```

Comment:
maybe use `verifyable` at where you set up these mocks and then just call `presentationManagerMock.verifyAll()` here?

Reply:
Changed to `verifyAll`